### PR TITLE
ARCO-193: remove unused columns add status history and last modified

### PR DESCRIPTION
**Pull request by pawellewandowski98**
## Description of Changes

- Remove unused columns: mined_ad and announced_at
- Add new columns: status_history and last_modified

## Linked Issues / Tickets

ARCO-193

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [ ] I have added new unit tests
- [ ] All tests pass locally
- [ ] I have tested manually in my local environment

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: internal/metamorph/store/postgresql/postgres_helpers.go [hunk 1]
```diff
@@ -207,6 +202,8 @@ func updateDoubleSpendRejected(ctx context.Context, rows *sql.Rows, tx *sql.Tx)
 		return nil
 	}
 
+	//rows.Close()
```
Comment (boecklim) [inline internal/metamorph/store/postgresql/postgres_helpers.go:205]: Remove comment

File: internal/metamorph/store/postgresql/postgres_helpers.go [hunk 2]
```diff
@@ -217,15 +214,17 @@ func updateDoubleSpendRejected(ctx context.Context, rows *sql.Rows, tx *sql.Tx)
 		return nil
 	}
 
+	//defer rows.Close()
```
Comment (boecklim) [inline internal/metamorph/store/postgresql/postgres_helpers.go:217]: Move after line 210 and uncomment

File: internal/metamorph/store/postgresql/postgres_test.go [hunk 3]
```diff
@@ -175,27 +175,27 @@ func TestPostgresDB(t *testing.T) {
...
 	unminedData := &store.StoreData{
-		RawTx:       make([]byte, 0),
-		StoredAt:    now,
-		AnnouncedAt: time.Date(2023, 10, 1, 12, 5, 0, 0, time.UTC),
-		Hash:        unminedHash,
-		Status:      metamorph_api.Status_SENT_TO_NETWORK,
-		LockedBy:    "metamorph-1",
+		RawTx:    make([]byte, 0),
+		StoredAt: now,
+		//AnnouncedAt: time.Date(2023, 10, 1, 12, 5, 0, 0, time.UTC),
```
Comment (boecklim) [inline internal/metamorph/store/postgresql/postgres_test.go:195]: Remove comment

File: internal/metamorph/store/postgresql/postgres_test.go [hunk 4]
```diff
@@ -175,27 +175,27 @@ func TestPostgresDB(t *testing.T) {
 	now := time.Date(2023, 10, 1, 14, 25, 0, 0, time.UTC)
 	minedHash := testdata.TX1Hash
 	minedData := &store.StoreData{
-		RawTx:        make([]byte, 0),
-		StoredAt:     now,
-		AnnouncedAt:  time.Date(2023, 10, 1, 12, 5, 0, 0, time.UTC),
-		MinedAt:      time.Date(2023, 10, 1, 12, 10, 0, 0, time.UTC),
+		RawTx:    make([]byte, 0),
+		StoredAt: now,
+		//StatusHistory: nil,
```
Comment (boecklim) [inline internal/metamorph/store/postgresql/postgres_test.go:180]: Remove comment

File: internal/metamorph/store/postgresql/postgres_test.go [hunk 5]
```diff
@@ -641,7 +641,7 @@ func TestPostgresDB(t *testing.T) {
 		minedReturned, err := postgresDB.Get(ctx, unminedHash[:])
 		require.NoError(t, err)
 		minedReturned.Status = metamorph_api.Status_MINED
-		minedReturned.MinedAt = now
+		//minedReturned.MinedAt = now
```
Comment (boecklim) [inline internal/metamorph/store/postgresql/postgres_test.go:644]: Remove comment

File: internal/metamorph/store/postgresql/postgres_test.go [hunk 6]
```diff
@@ -679,7 +679,7 @@ func TestPostgresDB(t *testing.T) {
 		dataReturned, err := postgresDB.Get(ctx, unminedHash[:])
 		require.NoError(t, err)
 		unmined.Status = metamorph_api.Status_MINED
-		unmined.MinedAt = now
+		//unmined.MinedAt = now
```
Comment (boecklim) [inline internal/metamorph/store/postgresql/postgres_test.go:682]: Remove comment

**Review by boecklim — APPROVED** [review]

File: internal/metamorph/store/postgresql/postgres_helpers.go (hunk 1 above)
Comment (pawellewandowski98) [reply internal/metamorph/store/postgresql/postgres_helpers.go]: Done

File: internal/metamorph/store/postgresql/postgres_helpers.go (hunk 2 above)
Comment (pawellewandowski98) [reply internal/metamorph/store/postgresql/postgres_helpers.go]: Done

File: internal/metamorph/store/postgresql/postgres_test.go (hunk 3 above)
Comment (pawellewandowski98) [reply internal/metamorph/store/postgresql/postgres_test.go]: Done

File: internal/metamorph/store/postgresql/postgres_test.go (hunk 4 above)
Comment (pawellewandowski98) [reply internal/metamorph/store/postgresql/postgres_test.go]: Done

File: internal/metamorph/store/postgresql/postgres_test.go (hunk 5 above)
Comment (pawellewandowski98) [reply internal/metamorph/store/postgresql/postgres_test.go]: Done

File: internal/metamorph/store/postgresql/postgres_test.go (hunk 6 above)
Comment (pawellewandowski98) [reply internal/metamorph/store/postgresql/postgres_test.go]: Done

File: internal/metamorph/store/postgresql/postgres_test.go [hunk 7]
```diff
@@ -177,25 +177,23 @@ func TestPostgresDB(t *testing.T) {
...
 	}
 
 	unminedHash := testdata.TX1Hash
 	unminedData := &store.StoreData{
-		RawTx:       make([]byte, 0),
-		StoredAt:    now,
-		AnnouncedAt: time.Date(2023, 10, 1, 12, 5, 0, 0, time.UTC),
-		Hash:        unminedHash,
-		Status:      metamorph_api.Status_SENT_TO_NETWORK,
-		LockedBy:    "metamorph-1",
```
Comment (kuba-4chain) [inline internal/metamorph/store/postgresql/postgres_test.go:198]: I see you removed `announcedAt` from every test, but you didn't add the new statusHistory to any unit test. Please add it to one, or possibly to couple of tests to see if everything is working correctly with parsing, saving, etc. 🙏

File: internal/metamorph/store/postgresql/postgres.go [hunk 8]
```diff
@@ -332,6 +340,8 @@ func (p *PostgreSQL) Set(ctx context.Context, value *store.StoreData) error {
 		value.RawTx,
 		p.hostname,
 		value.LastSubmittedAt,
+		statusHistoryData,
+		value.LastModified,
```
Comment (kuba-4chain) [inline internal/metamorph/store/postgresql/postgres.go:344]: Shouldn't we put `time.Now()` as `lastModified` field in each `Set...` query?

**Pull request merged**